AppOne/AppOne/DataServiceLayer/ServiceLayer.swift

```diff
@@ -0,0 +1,68 @@
+//
+//  ServiceLayer.swift
+//  AppOne
+//
+//  Created by Asapu, Ajaikumar on 13.12.22.
+//
+
+import Foundation
+
+protocol ServiceCalls {
+    func fetchToDOs(completion: @escaping (Result<[ToDos], ErrorType>) -> Void)
+    func fetchUsers() async throws -> [Users]
+    func fetchPhotos() async throws -> [Photos]
+}
+
+class ServiceLayer: ServiceCalls {
+
+    func fetchToDOs(completion: @escaping (Result<[ToDos], ErrorType>) -> Void) {
+        if let todosUrl = URL(string: "https://jsonplaceholder.typicode.com/todos") { URLSession.shared.dataTask(with: todosUrl) { (data, response, error) in
+            if error != nil {
+                completion(.failure(.serviceError))
+                return
+            }
+
+            if let data = data {
+                let todosData = try JSONDecoder().decode([ToDos].self, from: data)
+                completion(.success(todosData))
+                return
+            }
```

Comment:
better one guard for both error cases and the success one without intentation for better readibility